webauthn-authenticator-rs/src/ctap2/ctap21.rs

```diff
@@ -67,240 +60,118 @@ impl<'a, T: Token, U: UiCallback> Ctap21Authenticator<'a, T, U> {
-        friendly_name: String,
+        new_min_pin_length: Option<u32>,
+        min_pin_length_rpids: Vec<String>,
+        force_change_pin: Option<bool>,
     ) -> Result<(), WebauthnCError> {
-        // TODO: handle CTAP_2_1_PRE version too
-        self.check_fingerprint_support().await?;
-        let friendly_name = Some(self.check_friendly_name(friendly_name).await?);
-        self.bio(BioSubCommand::FingerprintSetFriendlyName(TemplateInfo {
-            id,
-            friendly_name,
-        }))
-        .await?;
-        Ok(())
+        self.config(
+            ConfigSubCommand::SetMinPinLength(SetMinPinLengthParams {
+                new_min_pin_length,
+                min_pin_length_rpids,
+                force_change_pin,
+            }),
+            false,
+        )
+        .await
     }
 
-    /// Removes an enrolled fingerprint.
-    pub async fn remove_fingerprint(&mut self, id: Vec<u8>) -> Result<(), WebauthnCError> {
-        // TODO: handle CTAP_2_1_PRE version too
-        self.check_fingerprint_support().await?;
-
-        self.bio(BioSubCommand::FingerprintRemoveEnrollment(id))
-            .await?;
-
-        Ok(())
+    /// Returns `true` if the authenticator supports
+    /// [enterprise attestation][0].
+    ///
+    /// # See also
+    ///
+    /// * [`enable_enterprise_attestation()`][Self::enable_enterprise_attestation]
+    ///
+    /// [0]: https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-errata-20220621.html#sctn-feature-descriptions-enterp-attstn
+    #[inline]
+    pub fn supports_enterprise_attestation(&self) -> bool {
+        self.info.supports_enterprise_attestation()
     }
 
-    /// Removes multiple enrolled fingerprints.
+    /// Enables the [Enterprise Attestation][0] feature.
     ///
-    /// **Warning:** this is not an atomic operation. If any command fails,
-    /// further processing will stop, and the request may be incomplete.
+    /// This is only available on authenticators which support
+    /// [configuration][Self::supports_config] and
+    /// [enterprise attestation][Self::supports_enterprise_attestation].
     ///
-    /// Call [Self::list_fingerprints] to check what was actually done.
-    pub async fn remove_fingerprints(&mut self, ids: Vec<Vec<u8>>) -> Result<(), WebauthnCError> {
-        let (iface, pin_uv_auth_token) = self
-            .get_pin_uv_auth_session(Permissions::BIO_ENROLLMENT, None, false)
-            .await?;
-
-        for id in ids {
-            self.bio_with_session(
-                BioSubCommand::FingerprintRemoveEnrollment(id),
-                iface.as_ref(),
-                pin_uv_auth_token.as_ref(),
-            )
-            .await?;
+    /// [0]: https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-errata-20220621.html#sctn-feature-descriptions-enterp-attstn
+    pub async fn enable_enterprise_attestation(&mut self) -> Result<(), WebauthnCError> {
```

Comment:
Also curious if we have any tokens that do this.

Reply:
I don't think we do; but this is also a copy-over of existing functionality.